homeassistant/components/barry/sensor.py

```diff
@@ -0,0 +1,63 @@
+"""Platform for sensor integration."""
+from datetime import timedelta
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(hours=1)
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Barry sensor."""
+    barry_connection = hass.data.get(DOMAIN)
+    price_code = hass.data.get("price_code")
+    async_add_entities([BarrySensor(barry_connection, price_code)], True)
+
+
+class BarrySensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, barry_home, price_code):
+        """Initialize the sensor."""
+        self._barry_home = barry_home
+        self.price_code = price_code
+        self._state = None
+        self._name = "Barry"
+        self._device_state_attributes = {}
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._device_state_attributes
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"Electricity price {self._name}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
```

Comment:
```suggestion
```

Reply:
We need to display the friendly name using "name" property and need to display the current hour spot price in the state column of "Current Entities". Therefor, we need both the properties to make this integraton workds.

Reply:
You don't need these to achieve that:

<https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes>